src/files/FilesPage.js

```diff
@@ -51,72 +35,35 @@ class FilesPage extends React.Component {
     doFilesMakeDir: PropTypes.func.isRequired
   }
 
-  state = defaultState
-
-  resetState = (field) => this.setState({ [field]: defaultState[field] })
-
-  setTogglers = (toggleOne, toggleAll) => this.setState({ toggleOne, toggleAll })
+  state = {
+    downloadAbort: null,
+    downloadProgress: null
+  }
 
   makeDir = (path) => this.props.doFilesMakeDir(join(this.props.files.path, path))
 
-  navigate = (link) => {
-    const { doUpdateHash, files } = this.props
-
-    if (files.type === 'directory') {
-      this.state.toggleAll(false)
-    }
-
-    link = link.split('/').map(p => encodeURIComponent(p)).join('/')
-    doUpdateHash(`/files${link}`)
+  navigate = (path) => {
+    const link = path.split('/').map(p => encodeURIComponent(p)).join('/')
+    this.props.doUpdateHash(`/files${link}`)
   }
 
-  showRenameModal = ([file]) => {
-    this.setState({
-      rename: {
-        isOpen: true,
-        path: file.path,
-        filename: file.path.split('/').pop()
-      }
-    })
-  }
-
-  rename = async (newName) => {
-    const {filename, path} = this.state.rename
-    this.resetState('rename')
-
-    if (newName !== '' && newName !== filename) {
-      this.state.toggleOne(filename, false)
-      await this.props.doFilesMove(path, path.replace(filename, newName))
-      this.state.toggleOne(newName, true)
+  updateFiles = async () => {
+    if (this.props.ipfsReady) {
+      const path = decodeURIComponent(this.props.routeInfo.params.path)
```

Comment:
Let's move this to the redux bundle as a selector like `selectRoutePath`. I needed to do something similar in the ipld explorer: https://github.com/ipfs-shipyard/ipld-explorer/blob/76c6172899bda3fdd1b59be5684ced25d95ff371/src/bundles/explore.js#L29-L36

Reply:
Will do!